sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -90,6 +91,8 @@
    */
   private ColumnarBatch columnarBatch;
 
+  private MemoryMode memoryMode = null;
```

Comment:
I agree. IIUC, only [these three test suites](https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala#L37) requires case 1 without calling `PinitBatch(partitionColumns, partitionValues, isEnabledOffHeap)`.
If we solve them, we can remove `(memoryMode != null) ?` at line 203.

Case 2 is still required since line 213 calls `initBatch()` after calling `initBatch(partitionColumns, partitionValues, isEnabledOffHeap)`.